code/go/0chain.net/smartcontract/storagesc/allocation_pool.go

```diff
@@ -142,6 +144,47 @@ type allocationPool struct {
 	Blobbers          blobberPools     `json:"blobbers"`      //
 }
 
+func newAllocationPool(
+	t *transaction.Transaction,
+	alloc *StorageAllocation,
+	until common.Timestamp,
+	mintNewTokens bool,
+	balances chainState.StateContextI,
+) (*allocationPool, error) {
+	var err error
+	if !mintNewTokens {
+		if err = checkFill(t, balances); err != nil {
+			return nil, err
+		}
+	}
+
+	var ap allocationPool
+	var transfer *state.Transfer
+	if transfer, _, err = ap.DigPool(t.Hash, t); err != nil {
+		return nil, fmt.Errorf("digging write pool: %v", err)
+	}
+	if mintNewTokens {
+		balances.AddMint(&state.Mint{
+			Minter:     ADDRESS,
+			ToClientID: ADDRESS,
+			Amount:     state.Balance(t.Value),
+		})
+	} else {
+		if err = balances.AddTransfer(transfer); err != nil {
+			return nil, fmt.Errorf("adding transfer to write pool: %v", err)
+		}
+	}
+
+	// set fields
+	ap.AllocationID = alloc.ID
```

Comment:
It might have `nil pointer dereference` crash too.

Reply:
As before, if `sa` is nil then there is a serious coding error elsewhere and the chain is going to crash in the next few nanoseconds anyway.